[#12] 회원정보 수정 기능 구현

hanwix2 opened the pull request:
- 수정 가능 정보: 비밀번호, 전화번호, 주소
- 현재 세션의 User를 가져와 정보 수정 기능 구현
- 세션에 저장되는 데이터를 userId에서 UserSession 객체(필드: userId, userType)로 변경
- `LoginCheckInterceptor`: Interceptor로 특정 url에 접근 전 로그인이 되어있는지 확인 (annotation: `@UserAuthenticationNecessary`)
- `CurrentUserArgumentResolver`: 컨트롤러 메소드 파라미터 중 세션데이터(UserSession) 타입 확인 및 가져오기 (annotation: `@CurrentUser`)
- 테스트 코드 작성

hanwix2 committed:
[#12] 회원정보 수정 기능 구현

- 수정 가능 정보: 비밀번호, 전화번호, 주소
- 현재 세션의 User를 가져와 정보 수정 기능 구현
- 세션에 저장되는 데이터를 userId에서 UserSession 객체(필드: userId, userType)로 변경
- LoginCheckInterceptor: Interceptor로 특정 url에 접근 전 로그인이 되어있는지 확인 (annotation: @UserAuthenticationNecessary)
- CurrentUserArgumentResolver: 컨트롤러 메소드 파라미터 중 세션데이터(UserSession) 타입 확인 및 가져오기 (annotation: @CurrentUser)
- 테스트 코드 작성

f-lab-dev commented on src/main/java/com/show/showticketingservice/controller/MyPageController.java:
```diff
@@ -0,0 +1,29 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/my-infos")
```
신학님이 개발하신거랑 클래스가 겹치는것 같은데 어떻게 되신걸까요?

hanwix2 replied:
동일한 기능을 구현해야하는 부분이 있어 따로 회의해서 협의한 부분입니다!

hanwix2 replied:
이슈를 더 잘게 나누지 못했던 결과인거 같습니다..ㅠ

f-lab-dev commented on src/main/java/com/show/showticketingservice/tool/resolver/CurrentUserArgumentResolver.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Override
+    public boolean supportsParameter(MethodParameter methodParameter) {
+        return methodParameter.getParameterType().equals(UserSession.class);
```
타입이 `UserSession`면 무조건 대체시킨다라는건 `UserSession` 클래스를 사용하는 개발자가 의도하지 않더라도 대체되는 상황이 발생할 수 있지 않을까요?

tlsgkr7416 commented on src/main/java/com/show/showticketingservice/tool/interceptor/LoginCheckInterceptor.java:
```diff
@@ -0,0 +1,32 @@
+import javax.servlet.http.HttpServletResponse;
+
+@Component
+@Getter
```
여기 getter쓰신 이유가 있으신가요?

hanwix2 replied:
앗 필요없는 코드인데 이것저것 시도해보다가 삭제하는 것을 잊었습니다. 
감사합니다~

hanwix2 committed:
[#12] 아규먼트리졸버, 인터셉터 기능 관련 수정

- CurrentUserArgumentResolver - UserSession 타입 확인에서 CurrentUser 어노테이션 확인으로 변경
- 로그인 확인 인터셉터에서 사용하지 않는 getter 어노테이션 삭제

f-lab-dev reviewed (approved):
신학님 코드가 먼저 PR이 올라왔는 conflict 나는 부분은 잘 협의해서 합쳐주세요~
